### DISCUSSION TITLE
Quarkus application set build time config by default

### DISCUSSION THREAD

**Discussion by danieloh30** [quarkusio/quarkus#22071] [Q&A] [answered]
Hi All, 

Custom configuration using @ConfigProperty could be updated in Config Editor at DEV UI using the previous quarkus version(e.g., 2.2) but Quarkus 2.5.1 won't let me do it since the variable is a read-only config in "Build time config" now.

![image](https://user-images.githubusercontent.com/14066977/145420322-d961c72d-6d82-465c-8112-1234c03b0283.png)

Sample code:
```
    @ConfigProperty(name = "username")
    String username;
```

I was wondering if this is a new feature in 2.5.1 or a regression bug.

**Comment by danieloh30** [comment]
@maxandersen @stuartwdouglas @phillip-kruger

**Comment by stuartwdouglas** [comment, marked as answer]
This commit is what changed it: https://github.com/quarkusio/quarkus/commit/dc3171fb034295c60fc5cec9f110628029504023

Because of this code here: https://github.com/quarkusio/quarkus/blob/d7a2f2b40a56eb13c387ca9efd50497a707443ee/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/ConfigSourceName.java#L103

This is already fixed upstream, as I now ignore the 'editable' field, however that may have been a mistake, as you can't actually edit most config sources with an ordinal > 250.

@phillip-kruger Maybe we should restore the UI checking of the 'editable' field that I removed, and instead maybe hard code a check for the 'Build time config' source?

**Reply by phillip-kruger** [reply]
Sounds good to me.

**Reply by danieloh30** [reply]
This is fixed in `2.5.2`.
